src/index.ts

```diff
@@ -12,7 +12,17 @@ export function setupPlugin({ config, global }: PluginMeta) {
 // /* Runs on every event */
 export function processEvent(event: PluginEvent, { global }: PluginMeta) {
 
-    if (Math.random() < global.percentage / 100) {
+    // hash is a sha256 hash of the distinct_id represented in base 16
+    // We take the first 15 digits, convert this into an integer, 
+    // dividing by the biggest 15 digit, base 16 number to get a value between 0 and 1.
+    // This is stable, so a distinct_id that was allowed before will continue to be allowed,
+    // even if the percentage increases
+    const hash = createHash("sha256")
+        .update(event.distinct_id)
+        .digest("hex");
+    const decision_value = parseInt(hash.substring(0, 15), 16) / 0xfffffffffffffff;
+
+    if (decision_value < global.percentage / 100) {
```

Comment:
Could there be a once-in-a-2^16 case where `decision_value` is 1 and the percentage is `100` and `1 < 1` fails?

Reply:
Makes sense, good catch!